### PR TITLE
Add GitHub Actions tree node and children

### PR DESCRIPTION
**Pull request by MicroFish91**
Part 3 of https://github.com/microsoft/vscode-azurecontainerapps/issues/313#issuecomment-1518418209

### CODE REVIEW

**Comment by MicroFish91** [comment]
Side-note:  I'm not super satisfied with the current tree item context names used, I will probably do some overall renaming in a later PR, but didn't want to get carried away doing that in this one as it might touch a couple un-related tree items as well.

File: src/utils/actionUtils.ts [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+                colorId = 'testing.iconPassed';
+                break;
+        }
+    }
+
+    return new ThemeIcon(id, colorId ? new ThemeColor(colorId) : undefined);
+}
+
+// Status
+export function ensureStatus(data: { status: string | null }): Status {
```
Comment (bwateratmsft) [inline src/utils/actionUtils.ts:90]: This isn't too different from `nonNullProp`, can that be used instead?

File: src/utils/actionUtils.ts [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    if (Object.values(Status).includes(<Status>data.status)) {
+        return <Status>data.status;
+    } else {
+        throw new RangeError(localize('invalidStatus', 'Invalid status "{0}".', data.status));
+    }
+}
+
+// Helpers...
+function ensureConclusion(data: { conclusion: string | null }): Conclusion {
```
Comment (bwateratmsft) [inline src/utils/actionUtils.ts:99]: This isn't too different from `nonNullProp`, can that be used instead?

File: src/tree/gitHub/ActionTreeItem.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+            const { owner, name: repo } = gitUrlParse(this.actionWorkflowRuns.repository.html_url);
+            const getJobsParams: GetJobsParams = {
+                owner,
+                repo,
+                run_id: this.actionWorkflowRuns.id
+            };
+            return await getJobs(context, getJobsParams);
+        });
+
+        if (jobsData && jobsData.total_count > 0) {
```
Comment (bwateratmsft) [inline src/tree/gitHub/ActionTreeItem.ts:52]: What you have is valid, but another way to write it is this:

```suggestion
        if (jobsData?.total_count) {
```

You can keep whichever format you like better.

File: src/utils/actionUtils.ts (hunk 1 above)
Comment (MicroFish91) [reply src/utils/actionUtils.ts:90]: Good catch, I should be able to just use that + typecast

File: package.json [hunk 4]
```diff
@@ -384,6 +384,7 @@
         "dayjs": "^1.11.3",
         "dotenv": "^16.0.0",
         "fs-extra": "^8.1.0",
+        "git-url-parse": "^13.1.0",
```
[the remark this answers is not in the document]
Comment (MicroFish91) [reply package.json:387]: Yeah sure, I'll look into adding our own parsing utility

File: src/tree/gitHub/ActionTreeItem.ts (hunk 3 above)
Comment (MicroFish91) [reply src/tree/gitHub/ActionTreeItem.ts:52]: Definitely prefer your suggestion

**Pull request merged**